utils/initial_data_preparation.py

```diff
@@ -206,17 +201,30 @@ def get_base_repo_url(url):
     parsed_url = urlparse(url)
     path = parsed_url.path.strip("/")
 
-    # Check if the path ends with '.git' and strip it off
-    if path.endswith(".git"):
-        path = path[:-4]  # Remove the last 4 characters, '.git'
-
     parts = path.split("/")
+    print(f"parts: {parts}")
```

Comment:
Print statements seem more of a temporary debugging action than something for production use where logging is generally more suitable. No need to remove print statements now, more something for us to review from time to time. logging.debug might work well if we think this sort of information might be useful when other people are running the script.